src/clj/rems/application/model.clj

```diff
@@ -384,6 +212,65 @@
                "(not= " (:application/id application) " " (:application/id event) ")"))
   application)
 
+(def dynamic-workflow
+  (permissions/compile-rules
+   [{:permission :see-everything}
+    {:permission :application.command/accept-invitation}
+    {:permission :application.command/accept-licenses}
+    {:permission :application.command/add-licenses}
+    {:permission :application.command/add-member}
+    {:permission :application.command/change-resources}
+    {:permission :application.command/close}
+    {:permission :application.command/comment}
+    {:permission :application.command/copy-as-new}
+    {:permission :application.command/create}
+    {:permission :application.command/decide}
+    {:permission :application.command/invite-member}
+    {:permission :application.command/remark}
+    {:permission :application.command/remove-member}
+    {:permission :application.command/request-comment}
+    {:permission :application.command/request-decision}
+    {:permission :application.command/return}
+    {:permission :application.command/revoke}
+    {:permission :application.command/save-draft}
+    {:permission :application.command/submit}
+    {:permission :application.command/uninvite-member}
+    {:role :handler :permission :application.command/approve}
+    {:role :handler :permission :application.command/reject}]))
+
+(def bureaucratic-workflow
+  (permissions/compile-rules
+   [{:permission :see-everything}
+    {:permission :application.command/accept-invitation}
+    {:permission :application.command/accept-licenses}
+    {:permission :application.command/add-licenses}
+    {:permission :application.command/add-member}
+    {:permission :application.command/change-resources}
+    {:permission :application.command/comment}
+    {:permission :application.command/copy-as-new}
+    {:permission :application.command/create}
+    {:permission :application.command/invite-member}
+    {:permission :application.command/remark}
+    {:permission :application.command/remove-member}
+    {:permission :application.command/request-comment}
+    {:permission :application.command/request-decision}
+    {:permission :application.command/return}
+    {:permission :application.command/revoke}
+    {:permission :application.command/save-draft}
+    {:permission :application.command/submit}
+    {:permission :application.command/uninvite-member}
+    {:role :decider :permission :application.command/approve}
+    {:role :decider :permission :application.command/reject}]))
+
+(defn- calculate-permissions [application event]
+  (let [whitelist (case (get-in application [:application/workflow :workflow/type])
+                    :workflow/dynamic dynamic-workflow
+                    :workflow/bureaucratic bureaucratic-workflow
+                    :workflow/master master-workflow/whitelist)]
+    (-> application
+        (master-workflow/calculate-permissions event)
```

Comment:
is it weird that the master-workflow gets its own ns but the others don't?

Reply:
Master workflow is special. The others could be even just data in the config file.